tests/draft6/ref.json

```diff
@@ -328,5 +328,110 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "Location-independent identifier",
+        "schema": {
+            "$ref": "#foo",
+            "definitions": {
+                "A": {
+                    "$id": "#foo",
+                    "type": "integer"
+                }
+            }
+        },
+        "tests": [
+            {
+                "data": 1,
+                "description": "match",
+                "valid": true
+            },
+            {
+                "data": "a",
+                "description": "mismatch",
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "Canonical location-independent identifier",
+        "schema": {
+            "$ref": "http://localhost:1234/bar#foo",
+            "definitions": {
+                "A": {
+                    "$id": "http://localhost:1234/bar#foo",
+                    "type": "integer"
+                }
+            }
+        },
+        "tests": [
+            {
+                "data": 1,
+                "description": "match",
+                "valid": true
+            },
+            {
+                "data": "a",
+                "description": "mismatch",
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "Location-independent identifier with base URI change in subschema",
+        "schema": {
+            "$id": "http://localhost:1234/root",
+            "$ref": "http://localhost:1234/nested.json#foo",
+            "definitions": {
+                "A": {
+                    "$id": "nested.json",
+                    "definitions": {
+                        "B": {
+                            "$id": "#foo",
+                            "type": "integer"
+                        }
+                    }
+                }
+            }
+        },
+        "tests": [
+            {
+                "data": 1,
+                "description": "match",
+                "valid": true
+            },
+            {
+                "data": "a",
+                "description": "mismatch",
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "Location-independent identifiers should take precedent over regular fragments",
```

Comment:
I'm trying to figure out the failure mode of this test case.  Is there reason to think that an implementation might insert a `/` after the `#` and convert it into a pointer?

Reply:
At first I didn't notice the difference between JSON pointers which require a `/` and a regular plain fragment which doesn't. Now on second glance this test makes no sense and I'll remove it as it is only confusing and has no purpose.